teuthology/report.py

```diff
@@ -453,6 +490,164 @@ def delete_run(self, run_name):
+                machine_type=machine_type,
+            ))
+        else:
+            msg = response.text
+            self.log.error(
+                "PUT to {uri} failed with status {status}: {msg}".format(
+                    uri=uri,
+                    status=response.status_code,
+                    msg=msg,
+                ))
+
+        response.raise_for_status()
+    
+
+    def queue_stats(self, machine_type):
+        uri = "{base}/queue/stats/".format(
+            base=self.base_uri
+        )
+        queue_info = {'machine_type': machine_type}
+        queue_json = json.dumps(queue_info)
+
+        headers = {'content-type': 'application/json'}
+        response = self.session.post(uri, data=queue_json, headers=headers)
+
+        if response.status_code == 200:
+            self.log.info("Successfully retrieved stats for queue {machine_type}".format(
+                machine_type=machine_type,
+            ))
+            return response.json()
+        else:
+            msg = response.text
+            self.log.error(
+                "POST to {uri} failed with status {status}: {msg}".format(
+                    uri=uri,
+                    status=response.status_code,
+                    msg=msg,
+                ))
+        response.raise_for_status()
+    
+    def queued_jobs(self, machine_type, user):
+        uri = "{base}/queue/queued_jobs/".format(
+            base=self.base_uri
+        )
+        request_info = {'machine_type': machine_type,
+                        'user': user}
+        request_json = json.dumps(request_info)
+        headers = {'content-type': 'application/json'}
+        response = self.session.post(uri, data=request_json, headers=headers)
+
+        if response.status_code == 200:
+            self.log.info("Successfully retrieved jobs for user {user}".format(
+                user=user,
+            ))
+            return response.json()
+        else:
+            msg = response.text
+            self.log.error(
+                "POST to {uri} failed with status {status}: {msg}".format(
+                    uri=uri,
+                    status=response.status_code,
+                    msg=msg,
+                ))
+        response.raise_for_status()
+
+
+def create_machine_type_queue(machine_type):
+    reporter = ResultsReporter()
+    if not reporter.base_uri:
+        return
+    reporter.create_queue(machine_type)
+
+
+def get_user_jobs_queue(machine_type, user):
+    reporter = ResultsReporter()
+    if not reporter.base_uri:
+        return
+    return reporter.queued_jobs(machine_type, user)
+
+
+def pause_queue(machine_type, paused_by, pause_duration):
+    reporter = ResultsReporter()
+    if not reporter.base_uri:
+        return
+    paused = True
+    reporter.update_queue(machine_type, paused, paused_by, pause_duration)
+    paused = False
+    timer = threading.Timer(int(pause_duration), reporter.update_queue, [machine_type, paused, paused_by])
```

Comment:
@djgalloway how useful is pausing for a duration, as opposed to a binary pause/unpause command?

Seems like this is equivalent to ```teuthology-queue pause ... ; sleep $duration; teuthology-queue unpause ...```, and doing it server side (in paddles or the dispatcher) would bring more complexity.

Reply:
Sorry, why is the queue getting paused?

Reply:
@djgalloway I'm asking about how you use the 'teuthology-queue pause' command - since the queue is being moved to the db, it'd be simpler to avoid implementing a 'pause duration' option, which beanstalkd provided.

I'm not sure if you or anyone else used the duration

Reply:
Oh.  I personally use it to pause the queue for a number of reasons.  I think the pause duration is in case whomever paused the queue forgets and then signs off for the day.